generator/src/generator.rs

```diff
@@ -17,12 +18,42 @@ use pest::unicode::unicode_property_names;
 use pest_meta::ast::*;
 use pest_meta::optimizer::*;
 
-pub fn generate(
+#[derive(Debug)]
+pub(crate) struct DocComment {
+    /// Multi-line grammar doc, (joined with `\n`)
+    ///
+    /// e.g.
+    ///
+    /// ```ignore
+    /// "grammar doc 1\ngrammar doc 2"
+    /// ```
+    grammar_doc: String,
+    /// HashMap rule name and doc comments (joined with `\n`)
+    ///
+    /// e.g.
+    ///
+    /// ```ignore
+    /// { "foo": "line doc 1\nline doc 2", "bar": "line doc 3" }
+    /// ```
+    line_docs: HashMap<String, String>,
+}
+
+impl DocComment {
+    pub fn new(grammar_doc: String, line_docs: HashMap<String, String>) -> Self {
```

Comment:
maybe `new` could take `pairs: Pairs<'_, Rule>` and call `consume_grammar_doc` / `consume_line_docs` (it could potentially fuse them in one iteration or still call them separately)?

Reply:
Do you means move `consume_grammar_doc` and `consume_line_docs` implementation into DocComment?

Reply:
How about this change?

https://github.com/pest-parser/pest/pull/765/commits/4d9d1af01251ec4dfb63e563656a024765ff23ea

Reply:
yes, that works to logically group those functions in a module